modules/nickserv/enforce.c

```diff
@@ -445,7 +445,7 @@ ns_cmd_regain(struct sourceinfo *si, int parc, char *parv[])
 	}
 	if ((si->smu == mn->owner) || verify_password(mn->owner, password))
 	{
-		if (MOWGLI_LIST_LENGTH(&mn->owner->logins) >= me.maxlogins)
+		if (user_loginmaxed(si->smu))
```

Comment:
Shouldn't this be testing `mn->owner` instead of `si->smu->owner` because while in the first case (`si->smu == mn->owner`) these are equal in the second case they won't be (`verify_password(mn->owner, password)`)?

```suggestion
		if (user_loginmaxed(mn->owner))
```